wsidicom/instance/dataset.py

```diff
@@ -550,11 +550,14 @@ def matches_instance(self, other_dataset: "WsiDataset") -> bool:
         bool
             True if same instance.
         """
+
         return (
             self.uids == other_dataset.uids
             and self.image_size == other_dataset.image_size
             and self.tile_size == other_dataset.tile_size
             and self.tile_type == other_dataset.tile_type
+            and (getattr(self, 'TotalPixelMatrixOriginSequence', None) ==
+                 getattr(other_dataset, 'TotalPixelMatrixOriginSequence', None))
         )
 
```

Comment:
Nice check, as this will pass if both dataset (incorrect) misses total pixel matrix origin sequence.